sonar-plugin-api/src/main/java/org/sonar/api/ce/posttask/PostProjectAnalysisTaskTester.java

```diff
@@ -0,0 +1,530 @@
+    }
+
+    public QualityGateBuilder setName(String name) {
+      this.name = requireNonNull(name, NAME_CAN_NOT_BE_NULL);
+      return this;
+    }
+
+    public QualityGateBuilder setStatus(QualityGate.Status status) {
+      this.status = requireNonNull(status, STATUS_CAN_NOT_BE_NULL);
+      return this;
+    }
+
+    public QualityGateBuilder add(QualityGate.Condition condition) {
+      conditions.add(requireNonNull(condition, "condition can not be null"));
+      return this;
+    }
+
+    public QualityGateBuilder clearConditions() {
+      this.conditions.clear();
+      return this;
+    }
+
+    public QualityGate build() {
+      requireNonNull(id, ID_CAN_NOT_BE_NULL);
+      requireNonNull(name, NAME_CAN_NOT_BE_NULL);
+      requireNonNull(status, STATUS_CAN_NOT_BE_NULL);
+
+      return new QualityGate() {
+        @Override
+        public String getId() {
+          return id;
+        }
+
+        @Override
+        public String getName() {
+          return name;
+        }
+
+        @Override
+        public Status getStatus() {
+          return status;
+        }
+
+        @Override
+        public Collection<Condition> getConditions() {
+          return conditions;
+        }
+
+        @Override
+        public String toString() {
+          return "QualityGate{" +
+            "id='" + id + '\'' +
+            ", name='" + name + '\'' +
+            ", status=" + status +
+            ", conditions=" + conditions +
+            '}';
+        }
+      };
+    }
+  }
+
+  @Beta
+  public static final class ConditionBuilder {
+    private static final String METRIC_KEY_CAN_NOT_BE_NULL = "metricKey can not be null";
+    private static final String OPERATOR_CAN_NOT_BE_NULL = "operator can not be null";
+
+    private String metricKey;
+    private QualityGate.Operator operator;
+    private String errorThreshold;
+    private String warningThreshold;
+    private boolean onLeakPeriod;
+
+    private ConditionBuilder() {
+      // prevents instantiation outside PostProjectAnalysisTaskTester
+    }
+
+    public ConditionBuilder setMetricKey(String metricKey) {
+      this.metricKey = requireNonNull(metricKey, METRIC_KEY_CAN_NOT_BE_NULL);
+      return this;
+    }
+
+    public ConditionBuilder setOperator(QualityGate.Operator operator) {
+      this.operator = requireNonNull(operator, OPERATOR_CAN_NOT_BE_NULL);
+      return this;
+    }
+
+    public ConditionBuilder setErrorThreshold(@Nullable String errorThreshold) {
+      this.errorThreshold = errorThreshold;
+      return this;
+    }
+
+    public ConditionBuilder setWarningThreshold(@Nullable String warningThreshold) {
+      this.warningThreshold = warningThreshold;
+      return this;
+    }
+
+    public ConditionBuilder setOnLeakPeriod(boolean onLeakPeriod) {
+      this.onLeakPeriod = onLeakPeriod;
+      return this;
+    }
+
+    public QualityGate.Condition buildNoValue() {
+      checkCommonProperties();
+      return new QualityGate.Condition() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this anonymous class number of lines from 48 to at most 20, or make it a named class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1188)